src/target/riscv/riscv.c

```diff
@@ -5262,6 +5262,26 @@ COMMAND_HANDLER(handle_riscv_virt2phys_mode)
 	return ERROR_OK;
 }
 
+COMMAND_HANDLER(riscv_set_external_trigger)
+{
+	struct target *target = get_current_target(CMD_CTX);
+	RISCV_INFO(r);
+
+	if (CMD_ARGC != 1) {
+		LOG_ERROR("Command takes exactly 1 parameter.");
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+	int value = atoi(CMD_ARGV[0]);
+	if (value <= 0 || value > 16) {
+		LOG_ERROR("%s is not a valid integer argument for command.", CMD_ARGV[0]);
+		return ERROR_FAIL;
+	}
+
+	r->external_trigger = value;
```

Comment:
It would be safer to add some checks to prevent user from associating one trigger to multiple harts on same DM.